env/work/srw_python/uti_plot_matplotlib.py

```diff
@@ -104,10 +104,36 @@ def uti_plot2d1d(self, data, x_range, y_range, xc, yc, labels, _graphs_joined=Tr
         return self._maybe_savefig(fig)
 
     #def srw_ascii_plot(fname):
-    def uti_data_file_plot(self, _fname, _read_labels=1, _e=0, _x=0, _y=0, _graphs_joined=1, _traj_report=False, _traj_axis='x'):
+    def uti_data_file_plot(self, _fname, _read_labels=1, _e=0, _x=0, _y=0, _graphs_joined=1, _traj_report=False, _traj_axis='x',
+                           _scale='linear', _width_pixels=None):
         #data, mode, allrange = srw_ascii_load(fname)
         #data, mode, allrange, arLabels, arUnits = _file_load(_fname, _read_labels)
         data, mode, allrange, arLabels, arUnits = uti_plot_com.file_load(_fname, _read_labels, _traj_report, _traj_axis)
+        data = np.array(data)
+        if _scale != 'linear':
+            available_scales = ['log', 'log2', 'log10']
+            if _scale not in available_scales:
+                raise ValueError('Scale "{}" is not supported. Available scales: {}.'.format(_scale, ', '.join(available_scales)))
+            data[np.where(data <= 0.)] = 1.e-23
+            data = getattr(np, _scale)(data)
+        if _width_pixels:
+            try:
+                from scipy.ndimage import zoom
+                data = np.reshape(data, (allrange[5], allrange[8]), order='C')
```

Comment:
I corrected the ordering to be `C` instead of `F`. It does not affect the results since later it converts the data back to 1D representation using the same ordering. It can be useful for debugging/visualization of the intermediate results before resizing.